File: android/src/main/java/expo/modules/adaptiveglass/ExpoAdaptiveGlassSliderModule.kt
package expo.modules.adaptiveglass

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoAdaptiveGlassSliderModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoAdaptiveGlassSlider")

    View(GlassSliderView::class) {
      Events("onValueChange", "onSlidingComplete")

      Prop("value") { view: GlassSliderView, value: Double -> view.value = value.toFloat() }
      Prop("disabled") { view: GlassSliderView, value: Boolean -> view.disabled = value }
      Prop("fillColor") { view: GlassSliderView, value: Int? -> view.fillColor = value }
      Prop("lens") { view: GlassSliderView, value: Boolean -> view.lens = value }

      OnViewDidUpdateProps { view: GlassSliderView ->
        view.propsDidUpdate()
      }
    }
  }
}

File: ios/ExpoAdaptiveGlassSliderModule.swift
import ExpoModulesCore

public class ExpoAdaptiveGlassSliderModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoAdaptiveGlassSlider")

    View(GlassSliderView.self) {
      Events("onValueChange", "onSlidingComplete")

      Prop("value") { (view, value: Double) in view.value = CGFloat(value) }
      Prop("disabled") { (view, value: Bool) in view.disabled = value }
      Prop("fillColor") { (view, value: UIColor?) in view.fillColor = value }
      Prop("lens") { (view, value: Bool) in view.lens = value }

      OnViewDidUpdateProps { view in
        view.propsDidUpdate()
      }
    }
  }
}
